src/tracing/esp32_diagnostic_trace/Diagnostics.h

```diff
@@ -0,0 +1,135 @@
+/*
+ *
+ *    Copyright (c) 2024 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <lib/core/TLVCircularBuffer.h>
+
+namespace chip {
+namespace Tracing {
+
+namespace Diagnostics {
+
+// Diagnostic TAGs
+enum D_TAG
+{
+    TIMESTAMP = 0,
+    LABEL,
+    VALUE
+};
```

Comment:
```suggestion
enum class DiagTag : uint8_t
{
    kTimestamp = 0,
    kLabel,
    kValue,
};
```

just trying to align with code-style, and then the usage will be `DiagTag::kValue`, it makes reading the code easier as well.